src/graph.rs

```diff
@@ -32,6 +31,62 @@ use std::fmt;
 use std::rc::Rc;
 use std::sync::Arc;
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+pub struct Position {
+  /// The 0-indexed line index.
+  pub line: usize,
+  /// The 0-indexed character index.
+  pub character: usize,
+}
+
+impl Position {
+  fn from_pos(
+    parsed_source: &ParsedSource,
+    pos: deno_ast::swc::common::BytePos,
+  ) -> Self {
+    let line_and_column_index =
+      parsed_source.source().line_and_column_index(pos);
+    Self {
+      line: line_and_column_index.line_index,
+      character: line_and_column_index.column_index,
+    }
+  }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
+pub struct Range {
+  #[serde(skip_serializing)]
+  pub specifier: ModuleSpecifier,
```

Comment:
I combined `Span` and `Range` together to just have `Range`. The reason is to prevent confusion with swc's `Span`. Hopefully that's ok? If not, let me know of any suggestions.